src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/ValueTask.cs

```diff
@@ -571,6 +571,31 @@ public Task<TResult> AsTask()
         /// <summary>Gets a <see cref="ValueTask{TResult}"/> that may be used at any point in the future.</summary>
         public ValueTask<TResult> Preserve() => _obj == null ? this : new ValueTask<TResult>(AsTask());
 
+        /// <summary>Casts a <see cref="ValueTask{TResult}"/> to <see cref="ValueTask"/>.</summary>
+        internal ValueTask AsValueTask()
+        {
+            object? obj = _obj;
+            Debug.Assert(obj == null || obj is Task<TResult> || obj is IValueTaskSource<TResult>);
+
+            if (obj is null)
+            {
+                return default;
+            }
+
+            if (obj is Task<TResult> t)
+            {
+                return new ValueTask(t);
+            }
+
+            if (obj is IValueTaskSource vts)
+            {
+                // This assumes the token used with IVTS is the same as used with IVTS<TResult>.
+                return new ValueTask(vts, _token);
+            }
```

Comment:
nit: personal coding style preference, feel free to ignore it if you don't like it

```suggestion
            if (obj is null)
            {
                return default;
            }
            else if (obj is Task<TResult> t)
            {
                return new ValueTask(t);
            }
            else if (obj is IValueTaskSource vts)
            {
                // This assumes the token used with IVTS is the same as used with IVTS<TResult>.
                return new ValueTask(vts, _token);
            }
```

Reply:
One more thing: could we swap the cast to `Task<TResult>` and `IValueTaskSource`? It seems that `IValueTaskSource` would be more common and it would reduce the number of casts for `UnixFileStreamStrategy`

Reply:
> It seems that IValueTaskSource would be more common and it would reduce the number of casts for UnixFileStreamStrategy

It will be right now, since the method is internal and there's only one caller.  However, every other method on ValueTask does it in the order of value, task, IVTS, as that's expected to be the most common (the vast, vast majority of asynchronously completing ValueTasks are backed by tasks, because that's what you get for an asynchronously completing `async ValueTask` by default).  As such, for consistency with the rest of the implementation, I'm going to leave it with this ordering.